go/ct/spc/specification.go

```diff
@@ -785,7 +786,7 @@ func getAllRules() []Rule {
 		},
 		conditions: []Condition{
 			RevisionBounds(R13_Cancun, NewestSupportedRevision),
-			Eq(ReadOnly(), false),
+			IsNotReadOnly(),
```

Comment:
I might have never addressed this explicitly, but I never liked this pattern we established with the pairs of conditions like `IsCode` and `IsData`, or `IsStorageCold` and `IsStorageWarm`. I always thought, when everything is settled I would take a day or so and rewrite this into the form that `ReadOnly` has here, with an `Eq(ReadOnly(), false)` or `Eq(ReadOnly(), true)`. I think structurally this is the more concise form, with less types/code to be defined and tested, benefiting from composition.

Replacing `Eq(ReadOnly(), false)` here with an `IsNotReadOnly()` kinda goes against this idea. If the goal is to restrict the domain of test cases produced for the `ReadOnly` domain, I think this could be addressed differently.

The [ReadOnly](https://github.com/Fantom-foundation/Tosca/blob/dd9a9f0ee94d2f53198c33a8e79d7afbd8cfae3b/go/ct/rlz/expressions.go#L202) expression right now has the [boolDomain](https://github.com/Fantom-foundation/Tosca/blob/dd9a9f0ee94d2f53198c33a8e79d7afbd8cfae3b/go/ct/rlz/domains.go#L35) which when being [sampled](https://github.com/Fantom-foundation/Tosca/blob/dd9a9f0ee94d2f53198c33a8e79d7afbd8cfae3b/go/ct/rlz/domains.go#L46-L52) for test cases produces both cases -- `true` and `false`.

If for the static / read-only state the domain should be restricted, you could define a specialized domain
```
type readOnlyDomain struct {
	boolDomain
}

func (readOnlyDomain) Samples(value bool) []bool {
	return []bool{value}
}

func (readOnlyDomain) SamplesForAll(as []bool) []bool {
	return as
}
```
and use this domain for the `ReadOnly()` expression. This has the same effect of limiting the test cases to the value the `ReadOnly` property is restricted to.